### PR TITLE
Code coverage setup

### PR DESCRIPTION
**Pull request by floreks**
I've extended configuration to generate coverage reports for frontend code. After each travis build it's uploaded to codecov.io automatically. Locally it is generated after each `gulp frontend-test` execution in 2 formats: lcov and html.

I've added 1 frontend test just to see if coverage is updated correctly.

It may need some additional tweaking to exclude some files during generating report. For now I've excluded them in my account settings on codecov.

zrzut ekranu z 2015-12-23 11-30-32

Example coverage report from last commit on my `code-coverage` branch: https://codecov.io/github/floreks/dashboard?branch=code-coverage

### CODE REVIEW

File: build/karma.conf.js [hunk 1]
```diff
@@ -87,6 +96,9 @@ module.exports = function(config) {
       // Make 'import ...' statements relative to the following paths.
       paths: [conf.paths.frontendSrc, conf.paths.frontendTest],
       transform: [
+        // Browserify transform for the istanbul code coverage tool. Isparta istrumenter for ES6
+        // code coverage.
+        ['browserify-istanbul', {'instrumenter': require('isparta')}],
```
Comment (bryk) [inline build/karma.conf.js]: Use ES6 imports instead of require.

File: build/test.js [hunk 2]
```diff
@@ -67,10 +68,22 @@ function runProtractorTests(doneFn) {
...
+function uploadCoverageReport() {
+  gulp.src(path.join(conf.paths.coverageReport, 'lcov.info')).pipe(codecov());
+}
+
+/**
  * Runs once all unit tests of the application.
  */
 gulp.task('test', ['frontend-test', 'backend-test']);
 
+gulp.task('coverage-codecov-upload', uploadCoverageReport);
```
Comment (bryk) [inline build/test.js]: Can you inline the uploadCoverageReport function here? And move the documentation also here.

File: .travis.yml [hunk 3]
```diff
@@ -52,3 +52,4 @@ before_script:
   # This simplifies initial set up and provides a very consistent build and test environment.
   - docker pull golang:1.4
 script: ./node_modules/.bin/gulp check:create-cluster
+after_script: npm run codecov
```
Comment (bryk) [inline .travis.yml]: Hmm... You could just do `./node_modules/.bin/gulp coverage-codecov-upload`. It is consistent with the previous line. What do you think?

**Comment by bryk** [comment]
Hey, this is nice! Are you able to address my minor comments before you leave? I'd love to have this PR in :)

File: build/karma.conf.js (hunk 1 above)
Comment (floreks) [inline build/karma.conf.js]: I've tried that but couldn't make it work. Don't know why but when I pass imported isparta object here then it doesn't work. This is the only way I was able to make it work.

Comment (bryk) [inline build/karma.conf.js]: Ok, add a TODO and let's merge as it is

**Comment by floreks** [comment]
Updated according to your comments. PTAL

**Comment by floreks** [comment]
Can you rerun travis? It seems like CI problem. Travis passed on my repo branch build.

**Comment by bryk** [comment]
Rerun

**Comment by bryk** [comment]
Awesome, thank you!

**Pull request merged**